weather.py

```diff
@@ -1,98 +1,46 @@
+import json
 import requests
 from bs4 import BeautifulSoup
 
-# Function to fetch page content with error handling
-def get_page_content(url):
-    try:
-        page = requests.get(url)
-        page.raise_for_status()  # Raise HTTPError for bad responses
-        return page.content
-    except requests.exceptions.RequestException as e:
-        print(f"Error fetching page content: {e}")
-        return None
+class WeatherScraper:
+    def __init__(self, cities_file):
+        self.cities = self.load_cities(cities_file)
+
+    def load_cities(self, file_path):
+        with open(file_path, 'r') as file:
+            return json.load(file)
+
+    def get_page_content(self, url):
+        try:
+            page = requests.get(url)
+            page.raise_for_status()
+            return page.content
+        except requests.exceptions.RequestException as e:
+            print(f"Error fetching page content: {e}")
+            return None
+
+    def get_weather(self, city, lat, lon):
+        url = f"https://forecast.weather.gov/MapClick.php?lat={lat}&lon={lon}"
+        page_content = self.get_page_content(url)
+
+        if page_content is not None:
+            soup = BeautifulSoup(page_content, "html.parser")
+            div = soup.find('div', id="seven-day-forecast-container")
+            if div:
+                li = div.find('li', class_="forecast-tombstone")
+                if li:
+                    p = li.find('p', class_="short-desc")
+                    if p:
+                        return f"\t{city}: {p.get_text()}"
```

Comment:
This can me moved to another method `_extract_weather_from_page()`  which returns the weather condition or `None` if not found